projects/packages/publicize/src/class-rest-controller.php

```diff
@@ -453,7 +466,13 @@ public function update_publicize_connection( $request ) {
 			$body,
 			'wpcom'
 		);
-		return rest_ensure_response( $this->make_proper_response( $response ) );
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		global $publicize;
+		return rest_ensure_response( $publicize->get_connection_for_user( (int) $connection_id ) );
```

Comment:
I tested to mark a connection as shared as non-admin and this is what was the `$response` showed and it was not an array and not a `WP_Error`

```log
[body] => {"code":"authorization_required","message":"This user is not authorized to modify this publicize connection.","data":403}
    [response] => Array
        (
            [code] => 500
            [message] => Internal Server Error
        )
```

Reply:
Thus, the request was a `200` instead of `403` or `500`

Reply:
Good sleuthing! I will fix that

Reply:
I haven't managed to change this yet from the jetpack backend, may be we could handle it as a part of the error handling sub-task?